algorithms/unit_tests/TestStdAlgorithmsTeamAdjacentFind.cpp

```diff
@@ -55,7 +55,7 @@ struct TestFunctorA {
       case 0: {
         const auto it = KE::adjacent_find(member, KE::cbegin(myRowViewFrom),
                                           KE::cend(myRowViewFrom));
-        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+        Kokkos::single(Kokkos::PerTeam(member), [=, *this]() {
```

Comment:
`[=]` and `[=, *this]` are not equivalent.

`[=]` and `[=, this]` are equivalent, as `[=]` effectively captured the object by reference.

Reply:
Yes but I think what we want is *this because _this_ points to the host object and because of https://github.com/kokkos/kokkos/issues/695

Reply:
Per @nmm0 [here](https://github.com/DARMA-tasking/vt/issues/925#issuecomment-1259849888), it's possible to end up with warnings coming from compilers passed `-std=c++17` about restating defaults with an explicit capture clause. I don't recall how that got resolved there, but Nic might.

Reply:
Also, isn't one of those the expansion of `KOKKOS_CLASS_LAMBDA`, modulo `__host__ __device__` annotations?

Reply:
Not sure if the Kokkos macros changes based on the standard. 
But I think solving the warning can be done via ifdeffing it, however we need to resolve if we want to use *this or this. 
I was going off the wiki (see [here](https://kokkos.github.io/kokkos-core-wiki/ProgrammingGuide/HierarchicalParallelism.html#restricting-execution-to-a-single-executor)) but then Daniel's comments made me realize this is not totally correct

Reply:
Since this uses an explicit functor, rather than a lambda, we copy the whole class to the device anyway and it should be OK to just use `[=, this]` (but this raises warnings for C++17 with some compilers and we have seen problems in particular with MSVC). If changing the functor to a lambda, I could see that using `[=, this]` is problematic and I was preferring `[=, *this]` for consistency (and portability in that case). Ultimately, this is just a test, and copying any of the class members is cheap anyway.

Reply:
Technically in this place it doesn't matter: It must be correct with [=, *this] but if it is it's ok to just capture via [&] because it's a nested lambda. Since this is all tests I don't have a preference and I am happy to go with [=, *this] which is the closest to what the test intended.